src/controls/dynamicForm/DynamicForm.tsx

```diff
@@ -147,7 +147,12 @@ export class DynamicForm extends React.Component<
    * Default React component render method
    */
   public render(): JSX.Element {
-    const { customFormatting, fieldCollection, hiddenByFormula, infoErrorMessages, isSaving } = this.state;
+    let {fieldCollection} = this.state
+    const { customFormatting, hiddenByFormula, infoErrorMessages, isSaving } = this.state;
+
+    if (this.props.fieldOrder) {
+      fieldCollection = this.sortFields(fieldCollection, this.props.fieldOrder)
```

Comment:
Is there a reason to sort fields during rendering instead of doing it during mouting / `fieldOrder` prop update?

From my perspective, this could be done in the `getListInformation` method, triggered in mount / update event, to avoid unnecessary re-ordering and which is already doing all the work around list / list item / fields.